x/muxer/muxer.go

```diff
@@ -215,12 +215,19 @@ func (m *muxer) DiffConfig(ctx context.Context, req *rpc.DiffRequest) (*rpc.Diff
 		}
 	}
 
+	responses := asyncJoin(subs)
+	if resp := dominatingDiffResponse(responses); resp != nil {
+		return resp, nil
+	}
```

Comment:
The logic below to merge diff responses should handle a single response correctly (`merge({diff}) = diff`). If it doesn't, that indicates a bug in merge.

What happens when we remove `dominatingDiffResponse`?

Reply:
I think this PR makes DiffConfig return ErrUnimplemented instead of the previous empty responses and it seems like these get returned from the muxer on line 282

Reply:
Yeah at first I observed it was swallowing detailedDiff seemingly intentionally. This is because it was merging a real diff with an empty diff from PF and selecting the PF one incorrectly. I since switched PF to return NotImplemented instead of a dummy empty response. I think this is how it should have been. With this change to NotImplemented in place I think the muxer needs to treat NotImplemented as a special error. dominatingDiffResponse is one way to do it but not the only way.